### PR TITLE
expression: Fix code style according to goword

### PR DESCRIPTION
**Pull request by gaocegege**
ref #3120 

Signed-off-by: Ce Gao <ce.gao@outlook.com>

### CODE REVIEW

File: expression/builtin.go [hunk 1]
```diff
@@ -33,7 +33,8 @@ type baseBuiltinFunc struct {
 	deterministic bool
 }
 
-func newBaseBuiltinFunc(args []Expression, ctx context.Context) baseBuiltinFunc {
+// newBaseBuiltinFunc creates a new BaseBuildFunc.
+func newBaseBuiltinFunc(ctx context.Context, args []Expression) baseBuiltinFunc {
```
Comment (zhexuany) [inline expression/builtin.go]: please make ci happy

Comment (gaocegege) [inline expression/builtin.go]: Yeah I know, It is a PR in WIP.

Comment (gaocegege) [inline expression/builtin.go]: There is a decision to be determined: should we reorder the parameters.

Google suggestions that:

> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. The Context should be the first parameter, typically named ctx

ref https://golang.org/pkg/context/

Now context variable in TiDB is the last parameter :confused:

Comment (coocood) [inline expression/builtin.go]: We have filtered out `context.Context` warning in Makefile.

Comment (gaocegege) [inline expression/builtin.go]: Alright, I use VS Code to dev and return the issues.

Forget it, I'll revert them back.

**Comment by gaocegege** [comment]
PTAL

## Main Changes

1. Add comment `// eval evals a xxxx.` before `// See https://xxxx`
2. Fix some typos according to https://goreportcard.com/report/github.com/pingcap/tidb

**Comment by coocood** [comment]
Well done!
LGTM

**Pull request merged**